libc/unistd_fsdir.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libc-tests
+ *
+ * Testing unistd.h file/fs/directory related functions
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: 
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+
+#include <unity_fixture.h>
+
+#include "common.h"
+
+
+FILE *fd;
+char buf[50];
+const char filename[] = "unistd_fsdir_file";
+const char directoryname[] = "unistd_fsdir_directory";
+int err;
+char toolongpath[PATH_MAX + 16];
+
+TEST_GROUP(unistd_fsdir);
+
+TEST_SETUP(unistd_fsdir)
+{
+	/* assert we are in root */
+	TEST_ASSERT_EQUAL_INT(0, chdir("/"));
+	TEST_ASSERT_NOT_NULL(getcwd(buf, 50));
+	TEST_ASSERT_EQUAL_STRING("/", buf);
+
+	/* clear buffer */
+	memset(buf, 0, 50);
+
+	/* clear/create file */
+	fd = fopen(filename, "w");
+	fclose(fd);
+
+	/* set too long path */
+	memset(toolongpath, 'a', PATH_MAX + 15);
+	toolongpath[PATH_MAX + 15] = '\0';
+}
+
+
+TEST_TEAR_DOWN(unistd_fsdir)
+{
+	TEST_ASSERT_EQUAL_INT(0, remove(filename));
+}
+
+
+TEST(unistd_fsdir, fsdir_getcwd)
+{
+	TEST_ASSERT_NOT_NULL(getcwd(buf, 50));
+	TEST_ASSERT_EQUAL_STRING("/", buf);
+
+	TEST_ASSERT_NULL(getcwd(buf, 0));
+	TEST_ASSERT_EQUAL_INT(EINVAL, errno);
+
+	TEST_ASSERT_NULL(getcwd(buf, 1));
+	TEST_ASSERT_EQUAL_INT(ERANGE, errno);
+}
+
+
+TEST(unistd_fsdir, fsdir_chdir_todir)
+{
+	/* test chdir to some directory */
+	TEST_ASSERT_EQUAL_INT(0, chdir("/dev"));
+	TEST_ASSERT_NOT_NULL(getcwd(buf, 50));
+	TEST_ASSERT_EQUAL_STRING("/dev", buf);
+
+	/* test chdir back to root */
+	TEST_ASSERT_EQUAL_INT(0, chdir("/dev"));
```

Comment:
Why chdir to /dev again? I guess it's to check if /dev -> /dev returns zero, but it shouldn't be then under the comment about returning to root

Reply:
Yes, you are right, there were two tests bound into one and comments do not illustrate the tests behavior properly. Added missing comment and separation. Thanks!